loki/transform/transform_inline.py

```diff
@@ -210,15 +213,22 @@ def inline_member_routine(routine, member):
     def _map_unbound_dims(var, val):
         """
         Maps all unbound dimension ranges in the passed array value `val` with the
-        indices from the local variable `var`. It returns the re-mapped symbol.
+        indices from the local variable `var`.  It returns the updated tuple of dimensions.
 
         For example, mapping the passed array `m(:,j)` to the local
         expression `a(i)` yields `m(i,j)`.
         """
-        val_free_dims = tuple(d for d in val.dimensions if isinstance(d, sym.Range))
-        var_bound_dims = tuple(d for d in var.dimensions if not isinstance(d, sym.Range))
-        mapper = SubstituteExpressionsMapper(dict(zip(val_free_dims, var_bound_dims)))
-        return mapper(val)
+        new_dimensions = list(val.dimensions)
+        indices = []
+        for index, dim in enumerate(val.dimensions):
```

Comment:
This helper method is called in a pretty tightly iterated place, so list comprehension should be preferred over <list>.append.